modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java

```diff
@@ -4023,18 +4023,20 @@ private void _validateAutoIncrementValue(
 	}
 
 	private void _validateExternalReferenceCode(
-			String externalReferenceCode, long companyId,
-			long objectDefinitionId, long objectEntryId)
-		throws PortalException {
+		String externalReferenceCode, long companyId, long objectDefinitionId,
+		long objectEntryId) {
 
 		ObjectEntry objectEntry = objectEntryPersistence.fetchByERC_C_ODI(
 			externalReferenceCode, companyId, objectDefinitionId);
 
 		if ((objectEntry != null) &&
 			(objectEntry.getObjectEntryId() != objectEntryId)) {
 
-			throw new ObjectEntryValuesException.MustNotBeDuplicate(
-				externalReferenceCode);
+			throw new DuplicateObjectEntryExternalReferenceCodeException(
+				StringBundler.concat(
+					"Duplicate object entry with external reference code ",
```

Comment:
@4lejandrito isn't the practice now not to put specific data in the message body for security reasons?

Reply:
This message is not exposed to the user, just to the logs. The message displayed to the user is [the one from the exception mapper](https://github.com/liferay/liferay-portal/blob/0bb9010ffb046dc55c0019a4a3307dbc894e8db2/modules/apps/portal-vulcan/portal-vulcan-impl/src/main/java/com/liferay/portal/vulcan/internal/jaxrs/exception/mapper/DuplicateExternalReferenceCodeExceptionMapper.java#L32-L36).